src/uct/ib/ud/base/ud_ep.c

```diff
@@ -322,10 +322,11 @@ static void uct_ud_ep_timer(ucs_wtimer_t *self)
     if (diff > 3 * iface->tx.tick) {
         ucs_trace("scheduling resend now: %lu last_send: %lu diff: %lu tick: %lu",
                   now, last_send, diff, ep->tx.tick);
-        uct_ud_ep_ctl_op_add(iface, ep, UCT_UD_EP_OP_ACK_REQ);
         uct_ud_ep_ca_drop(ep);
         uct_ud_ep_resend_start(iface, ep);
-    } else if ((diff > iface->tx.tick) && uct_ud_ep_is_connected(ep)) {
+    }
+
+    if ((diff > iface->tx.tick) && uct_ud_ep_is_connected(ep)) {
```

Comment:
maybe we can place the if in line 322 under this 'if'?
also the comment below: ` /* Try to request ACK or NAK twice before going into full resend mode */`